### ISSUE TITLE
Consider adding ChromosomeMappings to Data Libs on http://usegalaxy.org

### ISSUE THREAD

**Issue by jennaj** [galaxyproject/galaxy#1477]
This is an interesting, curated, popular community data resource that covers many of the most commonly used genomes.  https://github.com/dpryan79/ChromosomeMappings

Pulling the data into a data lib then keeping it updated is the primary work. Secondary would be annotating Galaxy dbkeys vs external genome names and some basic testing at each update.

Votes are the best way to indicate interest in pursuing this. I belive it has been discussed, but is not a WIP yet. If a dup request, please link to active ticket then close this one out.

Bonus: Create a tool that uses these files as indexes to convert a range of file formats

Migrated from Trello https://trello.com/c/ASSc3im6

What do you think @dpryan79 ?

ping @martenson @blankenberg @bgruening

**Comment by dpryan79** [comment]
It seems like a nice idea to me. I unfortunately don't know enough about the Galaxy internals to handle the dbkey associations, or else I'd give it a go myself.

**Comment by jennaj** [comment]
@dpryan79 That's ok, we can try to see if the UIC will take on a tool development or I'll just put these in a library. Existing tools can do the conversions with a combination of tools. Once there, I can create a workflow to help users.

I really just wanted to know if you were OK with Galaxy incorporting the data. Sounds like a yes? Think will be so very useful for many.

Appreciate your getting back to me about the idea :) jen

**Comment by bgruening** [comment]
@jennaj can you sketch up a tool how you imagine it. The IUC will take a look at it :)

**Comment by dpryan79** [comment]
@jennaj: Absolutely, I need to put a "do whatever you want with this" license on that repository :)

**Comment by jennaj** [comment]
@dpryan79 thank you!
@bgruening great! I'll draft something we can all review and tune

**Comment by jennaj** [comment]
Hi @dpryan79 & @bgruening 

For how to model the tool, we could leverage from the current option for this in the gffread tool. Update is to convert more file formats and do that as a stand-alone tool. (Not that we couldn't offer this in really almost any tool, with indexed identifier swap files! - those with or without a genome selection option)

---

## <img width="694" alt="gffread_identifier_replace" src="https://cloud.githubusercontent.com/assets/8999735/12936458/c5e7440e-cf53-11e5-9ab3-49057c9c7878.png">
1. Data could be indexed, so that input is a choice "from the history" or from a build-in index. Same format as standard tools, for a consistent look. Other files can be in a library or better, we explain where the indexed ones are.
2. Could keep it very simple and avoid a bunch of warning and "how-to"  - OR I can write something up for that. My thoughts are that linking to @dpryan79's repository (which has really good documentation about content) is best since it keeps the wrapper isolated from small updates. https://github.com/dpryan79/ChromosomeMappings 
3. Why "b"? Static duplication of too much external help content seems high maintenance for the wrapper help area. Once new wiki up, I'll also create a support section that we can link in for even more help that is Galaxy-usage specific - not content, format and version help. This kind of help can also be updated without a repository update, in response to any issues users seem to find the most troublesome. Ideally would have the basic area set with a static link up in time for initial publish. There is really no way to keep users from mixing up genome versions, no matter how much it is explained. It is learned context, usually a bit painfully learned, by doing it wrong a few times, and discovering the problem after a bunch of wasted time/analysis. Everyone goes through that with bioinfo.

Let me know thoughts on this.
